docs/how-to/h-upgrade.md

```diff
@@ -0,0 +1,115 @@
+# How to upgrade between minor versions
+
+> **:warning: WARNING:** this is currently a feature in "edge" or "beta" risks channels. We do NOT recommend to use features in production that are not in the `stable` channel! Also, contact [Canonical Data Platform team](https://chat.charmhub.io/charmhub/channels/data-platform) if you are interested in the topic.
+
+Charm upgrades allow admin to upgrade both operator code (e.g. the revision used by the charm) and/or the workload version. Note that since the charm code pin a particular version of the workload, a charm upgrade may or may not involve also a workload version upgrade. In general, the following guide only applies for in-place upgrade that involve at most minor version upgrade of Kafka workload, e.g. between Kafka 3.4.x to 3.5.x. Major workload upgrades are generally **NOT SUPPORTED** and they should be carried out using out-of-place migrations. Please refer to the how-to guide about cluster migration [here](/t/charmed-kafka-how-to-cluster-migration/10951) for more information on how this can be achieved.
+
+We strongly recommend to **NOT** perform any other extraordinary operations on the Kafka cluster, while upgrading. As an examples, these may be (but not limited) to the following:
+1. Adding or removing units
+2. Creating or destroying new relations
+3. Changes in workload configuration
+4. Upgrading other connected applications (e.g. Zookeeper)
+
+The concurrency with other operations is not supported, and it can lead the cluster into inconsistent states.
+
+## Minor upgrade process overview
+
+When performing an in-place upgrade process, the full process is composed by the following high-level steps:
+
+1. **Collect** all necessary pre-upgrade information, necessary for a rollback (if ever needed)
+2. **Prepare** the charm for the in-place upgrade, by running some preparatory tasks 
+3. **Upgrade** the charm and/or the workload. Once started all units in a cluster will refresh the charm code and undergo a workload restart/update. The upgrade will be aborted if the unit upgrade has failed, requiring the admin user to rollback.
+4. **Post-upgrade checks** to make sure all units are in the proper state and the cluster is healthy.
+
+## Step 1: Collect
+
+The first step is to record the revisions of the running application, as a safety measure for a rollback action if needed. To accomplish this, simply run the `juju status` command and look for the revisions of the deployed Kafka and Zookeeper applications. You can also retrieve this with the following command:
+
+```shell
+KAFKA_CHARM_REVISION=$(juju status --format json | yq .applications.<KAFKA_APP_NAME>.charm-rev)
```

Comment:
**question:** Is `yq` shipped by default? Maybe footnote an install just in case.